### ISSUE TITLE
MobileBERT decoder capabilities

### ISSUE THREAD

**Issue by guillaume-be** [huggingface/transformers#8969]
The current input parameters for MobileBERT indicate that the model may be used in a decoder setting. However, the model architecture does not contain a cross-attention mechanism and several inputs to the model are effectively never used: `encoder_hidden_states` and `encoder_attention_mask`.

This can be seen in:
- https://github.com/huggingface/transformers/blob/de6befd41f3986c68f4af302761b627cb6519eb7/src/transformers/models/mobilebert/modeling_mobilebert.py#L247, where these 2 inputs are not used
- https://github.com/huggingface/transformers/blob/de6befd41f3986c68f4af302761b627cb6519eb7/src/transformers/models/mobilebert/modeling_mobilebert.py#L330, where these inputs are just passed to the previous forward function (where they have no impact)
- https://github.com/huggingface/transformers/blob/de6befd41f3986c68f4af302761b627cb6519eb7/src/transformers/models/mobilebert/modeling_mobilebert.py#L496, where these parameters are not used (not even passed to the `MobileBertAttention`)
- https://github.com/huggingface/transformers/blob/de6befd41f3986c68f4af302761b627cb6519eb7/src/transformers/models/mobilebert/modeling_mobilebert.py#L552 where they are passed to the `MobileBertLayer` described above (therefore without impact)
- https://github.com/huggingface/transformers/blob/de6befd41f3986c68f4af302761b627cb6519eb7/src/transformers/models/mobilebert/modeling_mobilebert.py#L847 where they will trigger some reshaping of the attention mask, but eventually not get used.

I believe these unused inputs make the code more difficult to follow and potentially misleading (I don't believe the model can actually be used as a decoder).

Would you be generally supportive of a cleanup of the MobileBERT architecture to reflect its current capabilities? I'd be happy to share a PR but I wanted to check your general thoughts on this.

Thank you,

### Who can help
 albert, bert, GPT2, XLM: @LysandreJik 
(did not find anyone for MobileBert? But this is relevant for package maintenance I believe you may be the right person for this)

**Comment by LysandreJik** [comment]
You are right, there are never used. These seem to have been added by mistake during the original implementation. I would be all for a cleanup, as long as it doesn't touch to anything other than these attributes.
